src/storage/weak.ts

```diff
@@ -0,0 +1,146 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Store, StorageIndex, StorageValue, StorageObject } from "storage";
+
+/**
+ * A weak implementation for the Store interface.
+ *
+ * This means the data saved in this store does not persist throughout application runs.
+ */
+class WeakStore implements Store {
+  private store: StorageObject;
+
+  constructor() {
+    this.store = {};
+  }
+
+  /**
+   * Gets the value recorded to the given index on the store.
+   *
+   * @param index [optional] The index  of the entry we want to delete.
+   *
+   * @returns The value found for the given index on the storage.
+   *          If no index is given, the whole store is returned.
+   */
+  get(index?: StorageIndex): Promise<StorageValue> {
+    return new Promise(resolve => {
+      if (!index) return this.store;
+
+      let target: StorageValue = this.store;
+      for (const key of index) {
+        if (typeof target !== "undefined" && typeof target !== "string" && key in target) {
```

Comment:
```suggestion
        if (target !== undefined && typeof target !== "string" && key in target) {
```

I was recently bitten by checking against a string, since I had a typo in that string.

Reply:
Heh, but we are still checking agains a string right next to this first condition!

I check for `undefined` enough times that it may be good to acutally have a helper function such as `isUndefined` to check it instead of re-writing this everytime. Would you mind if I did that as a follow up though?

Reply:
Would a helper function be more concise than `obj === undefined`? I'm not sure we need a helper, but if you really want to add one, please add it in a follow-up.